packages/expo-tracking-transparency/plugin/src/withTrackingTransparency.ts

```diff
@@ -0,0 +1,39 @@
+import { ConfigPlugin, createRunOncePlugin } from '@expo/config-plugins';
+
+const pkg = require('expo-tracking-transparency/package.json');
+
+const DEFAULT_NSUserTrackingUsageDescription =
+  'This will allow the app to gather app-related data that can be used for tracking you or your device.';
+
+const withTrackingTransparency: ConfigPlugin<{
+  /**
+   * Sets the iOS `NSUserTrackingUsageDescription` permission message in `Info.plist`.
+   * Passing `false` will skip adding the permission.
+   * @default 'This will allow the app to gather app-related data that can be used for tracking you or your device.'
+   */
+  userTrackingPermission?: string | false;
+} | void> = (config, props) => {
+  config = withUserTrackingPermission(config, props);
+  return config;
+};
+
+export const withUserTrackingPermission: ConfigPlugin<{
+  userTrackingPermission?: string | false;
+} | void> = (config, { userTrackingPermission } = {}) => {
+  if (userTrackingPermission === false) {
+    if (config && config.ios && config.ios.infoPlist) {
```

Comment:
```suggestion
    if (config?.ios?.infoPlist) {
```

Reply:
avoiding using optional chaining to support users on node 13 and under

Reply:
Can you add a comment to upgrade this when we require Node 14+?

Reply:
Actually, `?.` should be fine -- this is written in TypeScript.

Reply:
updated